nitransforms/linear.py

```diff
@@ -446,40 +447,53 @@ def apply(
         if isinstance(spatialimage, (str, Path)):
             spatialimage = _nbload(str(spatialimage))
 
-        data = np.squeeze(np.asanyarray(spatialimage.dataobj))
-        output_dtype = output_dtype or data.dtype
+        # Avoid opening the data array just yet
+        input_dtype = spatialimage.header.get_data_dtype()
```

Comment:
This doesn't seem like a good idea. If you have `int16` on-disk, you're going to start resampling to `int16` without scale factors.

Reply:
Thanks! How would be the right way of not changing the input dtype unless an output_dtype is passed?

Reply:
[`nb.arrayproxy.get_obj_dtype(img.dataobj)`](https://nipy.org/nibabel/reference/nibabel.arrayproxy.html#get-obj-dtype) will tell you what the effective dtype of a data object is. It will tend to be `float64` for functional data, as that's what scaling does to data types, but for unscaled integer data, it will give you back `img.get_data_dtype()`.

Reply:
Something like this?
```suggestion
        input_dtype = nb.arrayproxy.get_obj_dtype(spatialimage.dataobj)
```

Reply:
Yes.